fsd_utils/config/commonconfig.py

```diff
@@ -3,6 +3,13 @@
 
 
 class CommonConfig:
+    def _resolve_secret_key(flask_env: str = None) -> str:
+        secret_key = os.getenv("SECRET_KEY", None)
+        if not secret_key:
+            if flask_env in ["dev", "test", "uat", "production"]:
```

Comment:
There's potentially something unintuitive about a common configuration thing having logic that is then environment specific, I guess it creates a cyclical dependency.

I'm not sure what I suggest here but I wonder if this is an indication that the secret key shouldn't live in the common config at all and rather be delegating to the app specific configuration, that or this could be doing a presence check that its in the environment and letting the apps deal with the consequences.

It's also very unlikely but I suppose it having the ability to set a default value if other settings are manipulated or environment names drift or similar might be undesirable behaviour in the future.

I don't think this is a blocker for the change and this addresses an issue thats known to have an impact right now, happy for it to just get in or to help think that through further depending on the required timeline.

Reply:
As dicsussed, think we should have a backlog item to tidy up config in general - will raise a ticket

Reply:
This is also slightly odd as post-award uses `prod` as that's what the copilot env is called, so I think things could slip through here. (But post-award doesn't use this commonconfig for this thing I believe).

Reply:
Can we say not in ['development', 'unit_test'] or similar?